uPortal-portlets/src/main/java/org/apereo/portal/portlets/favorites/FavoritesController.java

```diff
@@ -87,11 +99,32 @@ public String initializeView(Model model) {
         // viable configured marketplace.  This complexity may not be worth it.  Anyway it is not yet implemented.
         model.addAttribute("marketplaceFname", this.marketplaceFName);
 
-        List<IUserLayoutNodeDescription> collections =
+        final List<IUserLayoutNodeDescription> collections =
                 FavoritesUtils.getFavoriteCollections(userLayout);
         model.addAttribute("collections", collections);
 
-        List<IUserLayoutNodeDescription> favorites = FavoritesUtils.getFavoritePortlets(userLayout);
+        final List<IUserLayoutNodeDescription> rawFavorites = FavoritesUtils.getFavoritePortlets(userLayout);
+
+        /*
+         * Filter the collection by SUBSCRIBE permission.  (NOTE:  I wish this filtering were done
+         * by the Layout Manager, before now, but it isn't.)
```

Comment:
A couple of reasons (or maybe more)...

- Effort;  the `FavoritesController` appears to be closer to the information necessary to make these calculations
- Unintended consequences;  the Layout Manager (and the layout from it) is used extensively throughout the portal;  the "old school" (Simple) Layout Manager is actually the underpinnings of the DLM ("Distributed Layout Manager");  a change like that, if make too quickly, could have an adverse effect on fragments (and probably a few other things I can't immediately think of)